### PR TITLE
Removed .ini file usage, changed data_types for `--add-extra-file-info`

### LINKED ISSUE #24: Should --config-file and --user-label-file be the same thing?
In `pds4_create_xml_index`, currently we have two different ways of specifying configuration files. `-config-file` specifies things related to index file generation (like nillable elements and null constants) while `--user-label-file` specifies the contents of configurable portions of label files. On top of that, they are in different formations (.ini and YAML, respectively), requiring the user to learn two new file formats to use the tool.

- Should these really be separate files?
- Should these really have different formats?
- If combined, which format should be used?

### PR DESCRIPTION
**Pull request by esimpsons3ti**
**1. Configuration Files**

- `INI` format is no longer required for configuration files. `YAML` is the new standard. Functions parsing and using the configuration file contents have been changed to use this new format.
- `--config-file` can now use multiple configuration files by being invoked multiple times in the query:
```
pds4_create_xml_index . "**/**/*.xml" --config-file config_1.yaml --generate-label ancillary --config-file label_config.yaml
```
- `--label-user-input` has been removed, now that `--config-file` has expanded functionality.
- Documentation for `--label-user-input` has been removed.
- Documentation for `--config-file` has been updated.
- Unit tests for the configuration files have been updated.

**2. Data Types Update**
- The data types for `filename` and `filepath` have been updated. `filename` now has the pds:ASCII_File_Name data type. `filepath` now has the `pds:ASCII_File_Specification_Name` data type if the value is less than or equal to 255 characters. If it is longer, an error message is printed and the program exits.

Fixes #24

### CODE REVIEW

**Review by rfrenchseti — CHANGES_REQUESTED** [review]

File: pds4indextools/pds4_create_xml_index.py [hunk 1]
```diff
@@ -259,12 +254,12 @@ def filter_dict_by_glob_patterns(input_dict, glob_patterns, valid_add_extra_file
...
-        default_config_file=Path(__file__).resolve().parent/'pds4indextools.ini',
-        specified_config_file=None):
+        default_config_file=Path(__file__).resolve().parent/'pds4indextools.yaml',
+        specified_config_files=None):
     """
     Create a config object from a given configuration file.
 
-    This will always load in the default configuration file 'pds4indextools.ini'. In the
+    This will always load in the default configuration file 'pds4indextools.yaml'. In the
     event a specified configuration file is given, the contents of that file will
```
[the remark this answers is not in the document]
Comment (esimpsons3ti) [reply pds4indextools/pds4_create_xml_index.py:263]: Override or overwrite?

[the remark this answers is not in the document]
Comment (rfrenchseti) [reply pds4indextools/pds4_create_xml_index.py:263]: You could argue either way, but normally "override" means to take a part of something that's already there and change its behavior. So you have an existing configuration with many parameters, and you're overriding one of them. "Overwrite" means to take an existing configuration (or file) and replace it completely, not keeping what was there before. So I would use "override" in this case. In the previous case where we were replacing an entire file with a new file, we were "overwriting" it.

[the remark this answers is not in the document]
Comment (esimpsons3ti) [reply pds4indextools/pds4_create_xml_index.py:263]: Thank you for clarifying.

File: pds4indextools/pds4indextools.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+- pds:ASCII_Date_YMD: {
```
[the remark this answers is not in the document]
Comment (esimpsons3ti) [reply pds4indextools/pds4indextools.yaml:1]: Oddly enough, `0001-01-01` is not interpreted as a string in the YAML file. It's as `datetime.date(1, 1, 1)`. Here's the failed unit test message:

```
 AssertionError: assert datetime.date(1, 1, 1) == '0001-01-01'
```

[the remark this answers is not in the document]
Comment (rfrenchseti) [reply pds4indextools/pds4indextools.yaml:1]: Ah OK. That's something to mention in the documentation.

**Review by rfrenchseti — CHANGES_REQUESTED** [review]
This is nicely done and I only have minor comments. Rather than trying to write them all out, it was faster simply to make them myself. Please take a look at the diffs in my commit and see if there's anything you disagree with. Here is a summary:

- Removed the "Indices and tables" section from the main documentation page because those pages don't actually contain anything.
- Fixed a typo and added a few clickable links to the README file.
- According to the YAML syntax, `2020-01-01T00:00:00` is also supposed to be a datetime object. I'm not sure if this isn't crashing the code simply because you don't have a unit test for it, or whether it's because it isn't being parsed as a datetime. Nevertheless, for safety and consistency I put all date-time-like strings in quotes.
- I fixed a couple of bad indents in `index_label_template_pds.xml` that were flagged by VS Code.
- I fixed a bug in the actual tool where if you didn't specify both `nillable` and `label-contents` in a config file, it would crash.
- I also moved that update code out of the try/except block so that only the config file loading was inside the exception trap.
- I made a few minor modifications to the tool's main RST page, including:
  - Adding an Introduction header so the table of contents would look nicer
  - Changing the heading level of "Configuration Files" and its subheadings so the table of contents would look nicer
  - Adding an introductory paragraph to Configuration Files
  - Removing "Modification_History" from the list of classes, since it's not actually used
  - Updating the default config file contents
  - Adding an example of a label config file
  - Making a few minor wording changes

If you have no objection to my changes, I will go ahead and approve this PR.

**Comment by esimpsons3ti** [comment]
I approve of your changes. Feel free to squash and merge.

**Review by rfrenchseti — APPROVED** [review]

**Pull request merged**